cmd/generate.go

```diff
@@ -98,30 +90,212 @@ var generateCmd = &cobra.Command{
 					}
 				}
 			}
-		})
+		}()
 
-		run(config, newMigrationFilePath, migrationNumber)
+		err = run(config, newMigrationFilePath, migrationNumber)
+		if err != nil {
+			log.Fatalf("Failed to run: %v\n", err)
+		}
 
 		if flagDev {
 			for {
 				time.Sleep(time.Millisecond * 100)
-				run(config, newMigrationFilePath, migrationNumber)
+				err = run(config, newMigrationFilePath, migrationNumber)
+				if err != nil {
+					log.Fatalf("Failed to run: %v\n", err)
+				}
 			}
 		}
 	},
 }
 
-func run(config *internal.Config, newMigrationFilePath string, migrationNumber uint) {
-	updated, err := updateDiff(config, newMigrationFilePath, migrationNumber == 1)
-	if err != nil && !errors.Is(err, ErrInvalidModel) {
-		log.Fatalf("Failed to generate diff: %v\n", err)
+type PostgresFunction = func(targetContainerID, migrateContainerID string) error
+
+func withPostgresContainers(function PostgresFunction) error {
+	targetContainerID, err := internal.DockerRunPostgresContainer()
+	if err != nil {
+		return fmt.Errorf("failed to create target container: %w", err)
 	}
-	if updated {
-		err = writeTemplateFiles(config, migrationNumber)
-		if err != nil {
-			log.Fatalf("Failed to write template files: %v\n", err)
+	migrateContainerID, err := internal.DockerRunPostgresContainer()
+	if err != nil {
+		return fmt.Errorf("failed to create migrate container: %w", err)
+	}
+
+	defer func() {
+		internal.DockerKillContainer(targetContainerID)
+		internal.DockerKillContainer(migrateContainerID)
+	}()
+
+	return function(targetContainerID, migrateContainerID)
+}
+
+//nolint:gocognit
+func run(
+	config *internal.Config,
+	newMigrationFilePath string,
+	migrationNumber uint,
+) error {
+	return withPostgresContainers(func(targetContainerID, migrateContainerID string) error {
```

Comment:
The `withPostgresContainers` and `PostgresFunction` setup feels a bit too complex honestly. It's only being used once, within run().

I think it's safe to remove the wrapping and just move the content of `withPostgresContainers` into `run`?

Reply:
Yes, let's remove the wrapping. When I refactored it, it was more complex and it made sense to have this wrapping, but then I simplified it and forgot it.